File: 0083.删除排序链表中的重复元素/0083-删除排序链表中的重复元素.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        if (head == NULL || head->next == NULL)
            return head;
        ListNode* ptr = head->next;
        ListNode* pre = head;
        while(1){
            // cout<<ptr->val<<" "<<pre->val<<endl;
            while (pre->val == ptr->val){
                // cout<<"1"<<endl;
                ptr = ptr->next;  
                if (!ptr) break;
            }
            pre->next = ptr;
            pre = pre->next;
            if (!ptr) break;

        }
        return head;
        
    }
};

File: LeetCode-Python/0083.删除排序链表中的重复元素/0083-删除排序链表中的重复元素.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteDuplicates(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head:
            return head
        left, right = head, head
        while right:
            if left.val != right.val:
                left = left.next
                left.val = right.val
            right = right.next
        
        left.next = None
        return head